### PR TITLE
add common edge endpoints

### PR DESCRIPTION
**Pull request by mlfarrell**

### CODE REVIEW

File: modeling-cmds/src/def_enum.rs [hunk 1]
```diff
@@ -575,6 +575,15 @@ define_modeling_cmd_enum! {
             pub face_id: Uuid,
         }
 
+        /// Gets the shared edge between these two faces if it exists
+        #[derive(Debug, Clone, Serialize, Deserialize, JsonSchema, ModelingCmdVariant)]
+        pub struct Solid3dGetCommonEdge {
+            /// Which object is being queried.
+            pub object_id: Uuid,
+            /// The faces being queried (must be an array of length 2)
+            pub face_ids: Vec<Uuid>,
```
Comment (adamchalmers) [inline modeling-cmds/src/def_enum.rs:584]:
```suggestion
            pub face_ids: [Uuid; 2]
```

types > comments

File: modeling-cmds/src/ok_response.rs [hunk 2]
```diff
@@ -453,6 +453,13 @@ define_ok_modeling_cmd_response_enum! {
             pub edge: Option<Uuid>,
         }
 
+        /// The response from the `Solid3DGetCommonEdge` command.
+        #[derive(Debug, Serialize, Deserialize, Clone, JsonSchema, ModelingCmdOutput)]
+        pub struct Solid3dGetCommonEdge {
+            /// The UUID of the common edge.
```
Comment (adamchalmers) [inline modeling-cmds/src/ok_response.rs:459]:
```suggestion
            /// The UUID of the common edge, if any.
```

File: modeling-cmds/src/def_enum.rs (hunk 1 above)
Comment (mlfarrell) [reply modeling-cmds/src/def_enum.rs:584]: whoa.  didn't know the API could handle statically sized arrays.  :+1:

File: modeling-cmds/src/def_enum.rs [hunk 3]
```diff
@@ -575,6 +575,15 @@ define_modeling_cmd_enum! {
             pub face_id: Uuid,
         }
 
+        /// Gets the shared edge between these two faces if it exists
+        #[derive(Debug, Clone, Serialize, Deserialize, JsonSchema, ModelingCmdVariant)]
+        pub struct Solid3dGetCommonEdge {
+            /// Which object is being queried.
+            pub object_id: Uuid,
+            /// The faces being queried (must be an array of length 2)
```
Comment (adamchalmers) [inline modeling-cmds/src/def_enum.rs:583]:
```suggestion
            /// The faces being queried
```

**Review by adamchalmers — CHANGES_REQUESTED** [review]

**Review by adamchalmers — APPROVED** [review]

**Pull request merged**